src/stores/announcements/AnnouncementStore.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+import type { AnnouncementState } from '@/types/announcement';
+import type { StateCreator } from 'zustand';
+import { persist } from 'zustand/middleware';
+import { shallow } from 'zustand/shallow';
+import { createWithEqualityFn } from 'zustand/traditional';
+
+const defaultAnnouncements = {
+  dismissedAnnouncements: [],
+  lastFetchDate: null,
+};
+
+export const useAnnouncementStore = createWithEqualityFn<AnnouncementState>(
+  persist(
+    (set, get) => ({
+      ...defaultAnnouncements,
+
+      dismissAnnouncement: (documentId: string) => {
+        const dismissedAnnouncements = get().dismissedAnnouncements;
+        if (!dismissedAnnouncements.includes(documentId)) {
+          set({
+            dismissedAnnouncements: [...dismissedAnnouncements, documentId],
+          });
+        }
+      },
```

Comment:
Use an atomic update, something like:

```suggestion
      dismissAnnouncement: (documentId: string) => 
         set((state) => { dismissedAnnouncements: uniq([...state.dismissedAnnouncements, documentId])}
```